diesel/src/pg/connection/raw.rs

```diff
@@ -32,6 +32,9 @@ impl RawConnection {
             }
             _ => {
                 let message = last_error_message(connection_ptr);
+				
+                unsafe { PQfinish(connection_ptr) }
```

Comment:
This should either have a comment pointing to some postgres documentation stating that `PQfinish` handles null pointers on it's own or a check for a null pointer. Also adding a comment why we need to call `PQfinish` here would be a good idea (maybe just copy the documentation snipped from the issue?)
```suggestion
                if !connection_ptr.is_null() {
                     // Note that even if the server connection attempt fails (as indicated by PQstatus), 
                     // the application should call PQfinish to free the memory used by the PGconn object.
                     // https://www.postgresql.org/docs/current/libpq-connect.html
                     unsafe { PQfinish(connection_ptr) }
                }
```

Reply:
Makes sense to me, committed the change